### PR TITLE
Move all asset paths in examples to constants at the top of the example

### PR DESCRIPTION
**Pull request by Aztro-dev**
# Objective

- (hopefully) Complete #13645

## Solution

- Add a ton of constants at the top of files that hold paths to things in the `Assets` folder
- Also added documentation explaining where the constants come from

## Testing

- Ran (almost) all of the changed examples, but I still need to test examples in folders that come after `gizmos` alphabetically (not that many changed examples)

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
@kornelski @Wuketuke I'd appreciate your review on this. I'll trigger a manual example screenshot run for this and report back.

File: examples/games/loading_screen.rs [hunk 1]
```diff
@@ -2,17 +2,9 @@
 use bevy::{ecs::system::SystemId, prelude::*};
 use pipelines_ready::*;
 
-// The way we'll go about doing this in this example is to
-// keep track of all assets that we want to have loaded before
-// we transition to the desired scene.
-//
-// In order to ensure that visual assets are fully rendered
```
Comment (alice-i-cecile) [inline examples/games/loading_screen.rs:9]: I don't think this comment should be removed in this PR.

File: examples/scene/scene.rs [hunk 2]
```diff
@@ -2,6 +2,10 @@
 use bevy::{prelude::*, tasks::IoTaskPool, utils::Duration};
 use std::{fs::File, io::Write};
 
+/// This example uses two RON files from the asset subdirectory
+const SCENE_FILE_PATH: &str = "scenes/load_scene_example.scn.ron";
+const NEW_SCENE_FILE_PATH: &str = "scenes/load_scene_example-new.scn.ron";
```
Comment (alice-i-cecile) [inline examples/scene/scene.rs:7]: This should have a helpful comment like before.

**Review by alice-i-cecile — COMMENTED** [review]
A couple of comments got lost, but otherwise looks great. I think this is a substantial improvement for new users, and I like the little comments a lot more than I expected to.

Once those are fixed up this LGTM (assuming the example run comes up clean).

File: examples/games/loading_screen.rs (hunk 1 above)
Comment (Aztro-dev) [reply examples/games/loading_screen.rs]: I don't exactly remember deleting this (in fact I vividly remember moving the constants above it), but I reinstated the comment in my most recent commit

**Comment by alice-i-cecile** [comment]
Some further discussion with @cart on this style [here](https://discord.com/channels/691052431525675048/743559241461399582/1255984903967871136).

**Review by cart — CHANGES_REQUESTED** [review]
Sorry for the "back and forth" here, but I think we shouldn't use this pattern in our examples unless the path is reused. This adds additional boilerplate to all of the examples. This makes them _appear_ bigger and more complicated (which we are very careful to avoid ... perception is important). But it also _actually_ makes them more complicated, by nature of moving information away from its usage context.

This seems largely motivated by making it easier for people trying to copy/paste example code to identify what is missing. This _is_ a valid thing to think about and try to resolve, but I don't think it dominates the "simplicity and legibility" priority. If someone tries to run an example with a missing asset, it will print an error message with the missing path.  I think that is already a reasonable solve for subset of people that encounter this issue.

There are a few cases where an asset path is used more than once in an example. In those cases, I think this change is well-motivated. Feel free to open another PR with those changes (if that is interesting to you).

Sorry again for the back and forth. This is my fault for not weighing in earlier.

**Comment by kornelski** [comment]
> If someone tries to run an example with a missing asset, it will print an error message with the missing path.

This message difficult to spot, because by default Bevy prints a bunch of other stuff (like system info, GPU config, gamepads). I've learned to ignore the log messages.

The examples generally ignore all errors when loading assets, and code for catching the errors would be different complication of the examples. 

Could Bevy fail louder on missing assets?

**Comment by cart** [comment]
>  I've learned to ignore the log messages.

Imo you should unlearn this. We try not to crash Bevy unless we absolutely have to. The log is where we print error messages (and warnings). Look for red and yellow as indicators that something is wrong.

> Could Bevy fail louder on missing assets?

Something we could do for examples is add some kind of "log renderer" dev tool, which defaults to hidden and triggers visibility when an error occurs.

**Pull request closed without merging**